src/Serilog.Sinks.OpenTelemetry/Sinks/OpenTelemetry/LogRecordBuilder.cs

```diff
@@ -28,24 +28,33 @@ public static LogRecord ToLogRecord(LogEvent logEvent, IFormatProvider? formatPr
 
         ProcessProperties(logRecord, logEvent);
         ProcessTimestamp(logRecord, logEvent);
-        ProcessMessage(logRecord, logEvent, formatProvider);
+        ProcessMessage(logRecord, logEvent, includedFields, formatProvider);
         ProcessLevel(logRecord, logEvent);
         ProcessException(logRecord, logEvent);
         ProcessIncludedFields(logRecord, logEvent, includedFields, activityContextCollector);
 
         return logRecord;
     }
 
-    public static void ProcessMessage(LogRecord logRecord, LogEvent logEvent, IFormatProvider? formatProvider)
+    public static void ProcessMessage(LogRecord logRecord, LogEvent logEvent, IncludedData includedFields, IFormatProvider? formatProvider)
     {
         var renderedMessage = CleanMessageTemplateFormatter.Format(logEvent.MessageTemplate, logEvent.Properties, formatProvider);
```

Comment:
Rendering of the message should move inside the first `if` block, since it allocates/requires cycles that are wasted when the rendered message is not included (the first arm of the `if` block will need a nested `if (renderedMessage.Trim())` inside it - but the noise seems worth it for the improvement in behavior).